src/JustEat.StatsD/EndpointLookups/DnsLookupIpEndpointSource.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Net;
+
+namespace JustEat.StatsD.EndpointLookups
+{
+    /// <summary>
+    /// lookup IPAddress using DNS to find the host's IP
+    /// </summary>
+    public class DnsLookupIpEndpointSource : IPEndPointSource
+    {
+        private readonly string _hostName;
+        private readonly int _port;
+
+        public DnsLookupIpEndpointSource(string hostName, int port)
+        {
+            _hostName = hostName;
+            _port = port;
+        }
+
+        public IPEndPoint Endpoint => new IPEndPoint(GetIpAddressOfHost(_hostName), _port);
```

Comment:
This feels more like a method than a property to me as it does stuff that's non-trivial, plus if you accessed the value via the property more than once in the same method you'd get different values, which might be unexpected.